### ISSUE TITLE
GNNs for imbalanced node classification

### ISSUE THREAD

**Issue by bsaghafi** [rusty1s/pytorch_geometric#2889]
### Discussed in https://github.com/rusty1s/pytorch_geometric/discussions/2886

<div type='discussions-op-text'>

<sup>Originally posted by **bsaghafi** July 21, 2021</sup>
When using very imbalanced data, my experience is that GNN methods like GraphSAGE and GCN perform poorly, although I am using class ratio to weight the loss function accordingly, but still the classifier only predicts the majority class. Is there any feature or method other than loss weighting that can be used here? 

For better context, my problem is a binary classification where the class ratio is 400:1. Also I am using the ROC AUC metric on the validation set to determine the best number of epochs to train. I have also tried other metrics such as PR AUC and f1-score.</div>

**Comment by rusty1s** [comment]
Closing this in favor of https://github.com/rusty1s/pytorch_geometric/discussions/2886
